### PR TITLE
DOC Ensures that `SimilarityEncoder` passes numpydoc validation

### PR DESCRIPTION
**Pull request by LilianBoulard**
Part of #345.

### CODE REVIEW

**Review by jovan-stojanovic — APPROVED** [review]
Thank you!

File: dirty_cat/_similarity_encoder.py [hunk 1]
```diff
@@ -221,57 +220,68 @@ class SimilarityEncoder(OneHotEncoder):
...
+        Encodes dirty categories (strings) by constructing latent topics
+        with continuous encoding.
     :class:`~dirty_cat.deduplicate` :
         Deduplicate data by hierarchically clustering similar strings.
 
+    Notes
+    -----
+    The functionality of :class:`~SimilarityEncoder` is easy to explain
+    and understand, but it is not scalable.
+    Instead, the :class:`~dirty_cat.GapEncoder` is usually recommended over it.
```
Comment (jovan-stojanovic) [inline dirty_cat/_similarity_encoder.py:276]:
```suggestion
    Instead, the :class:`~dirty_cat.GapEncoder` is usually recommended.
```

**Comment by jovan-stojanovic** [comment]
Is there a parameter of the encoder that we could easily remove to simplify or not?

**Comment by LilianBoulard** [comment]
Doesn't look like it to me.
I'll be merging this PR, we can address this question in another :)

**Comment by LilianBoulard** [comment]
Any idea on why the CI fails @jovan-stojanovic ?
```
Traceback (most recent call last):
  File "/home/circleci/project/examples/06_ken_embeddings_example.py", line 141, in <module>
    X_full = fa1.fit_transform(X)
  File "/home/circleci/project/miniconda/envs/testenv/lib/python3.8/site-packages/sklearn/utils/_set_output.py", line 142, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
  File "/home/circleci/project/dirty_cat/_feature_augmenter.py", line 219, in fit_transform
    return self.fit(X).transform(X)
  File "/home/circleci/project/miniconda/envs/testenv/lib/python3.8/site-packages/sklearn/utils/_set_output.py", line 142, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
  File "/home/circleci/project/dirty_cat/_feature_augmenter.py", line 190, in transform
    X = fuzzy_join(
  File "/home/circleci/project/dirty_cat/_fuzzy_join.py", line 250, in fuzzy_join
    distance, neighbors = neigh.kneighbors(main_enc.toarray(), return_distance=True)
  File "/home/circleci/project/miniconda/envs/testenv/lib/python3.8/site-packages/scipy/sparse/_compressed.py", line 1051, in toarray
    out = self._process_toarray_args(order, out)
  File "/home/circleci/project/miniconda/envs/testenv/lib/python3.8/site-packages/scipy/sparse/_base.py", line 1298, in _process_toarray_args
    return np.zeros(self.shape, dtype=self.dtype, order=order)
numpy.core._exceptions._ArrayMemoryError: Unable to allocate 129. GiB for an array with shape (16514, 1048576) and data type float64
```

**Comment by jovan-stojanovic** [comment]
Yes, this is a memory error that apparently appeared after merging #509 .. This is because I force the array to become dense.

**Comment by jovan-stojanovic** [comment]
Great, thank you!

**Pull request merged**